app/renderer/js/pages/preference/general-section.ts

```diff
@@ -93,7 +95,10 @@ export default class GeneralSection extends BaseSection {
 						<div class="setting-description">${t.__('Enable spellchecker (requires restart)')}</div>
 						<div class="setting-control"></div>
 					</div>
+					<div class="setting-row" id="spellcheck-langs"></div>
```

Comment:
You can add this - 
`"display:${process.platform === 'darwin' ? '' : 'none'}"``, this will take care of one of your ToDo below.